BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -158,34 +165,71 @@ public actor DefaultTabDataStore: TabDataStore {
         }
     }
 
-    // MARK: Saving Data
+    // MARK: - Saving Data
+
     public func saveWindowData(window: WindowData) async {
-        guard let windowSavingPath = self.windowURLPath(for: window.id, isBackup: false) else {
+        guard let windowSavingPath = self.windowURLPath(for: window.id, isBackup: false)
+        else {
             return
         }
+
+        if checkIfFileExistsAtPath(path: windowSavingPath) {
+            createWindowDataBackup(window: window, windowSavingPath: windowSavingPath)
+        } else {
+            if let windowDataDirectoryURL = windowDataDirectoryURL,
+               !self.checkIfFileExistsAtPath(path: windowDataDirectoryURL) {
+                self.createDirectoryAtPath(path: windowDataDirectoryURL)
+            }
+        }
+        await writeWindowDataToFileWithThrottle(window: window, path: windowSavingPath)
+    }
+
+    private func createWindowDataBackup(window: WindowData, windowSavingPath: URL) {
+        guard let backupWindowSavingPath = self.windowURLPath(for: window.id, isBackup: true),
+              let backupDirectoryPath = self.windowDataBackupDirectoryURL else {
+            return
+        }
+        if !self.checkIfFileExistsAtPath(path: backupDirectoryPath) {
+            self.createDirectoryAtPath(path: backupDirectoryPath)
+        }
         do {
-            if self.checkIfFileExistsAtPath(path: windowSavingPath) {
-                guard let backupWindowSavingPath = self.windowURLPath(for: window.id, isBackup: true), let backupDirectoryPath = self.windowDataBackupDirectoryURL else {
+            try FileManager.default.copyItem(at: windowSavingPath, to: backupWindowSavingPath)
+        } catch {
+            self.logger.log("Failed to create window data backup: \(error)", level: .debug, category: .tabs)
```

Comment:
nit: cleanup for file consistency based on other areas seen below
```suggestion
            self.logger.log("Failed to create window data backup: \(error)",
                            level: .debug,
                            category: .tabs)
```